Demo/Common/IntegrationExamples/Components/ApplePayComponentExample.swift

```diff
@@ -0,0 +1,129 @@
+            let component = try applePayComponent(from: session)
+            guard let componentViewController = component?.viewController else { return }
+            presenter?.present(viewController: componentViewController, completion: nil)
+            applePayComponent = component
+        } catch {
+            self.presentAlert(with: error)
+        }
+    }
+
+    internal func applePayComponent(from session: AdyenSession) throws -> ApplePayComponent? {
+        let paymentMethods = session.sessionContext.paymentMethods
+        guard let paymentMethod = paymentMethods.paymentMethod(ofType: ApplePayPaymentMethod.self),
+              let applePayPayment = try? ApplePayPayment(payment: ConfigurationConstants.current.payment,
+                                                         brand: ConfigurationConstants.appName)
+        else {
+            throw IntegrationError.paymentMethodNotAvailable(paymentMethod: CardPaymentMethod.self)
+        }
+        var config = ApplePayComponent.Configuration(payment: applePayPayment,
+                                                     merchantIdentifier: ConfigurationConstants.applePayMerchantIdentifier)
+        config.allowOnboarding = true
+        config.shippingType = .delivery
+        config.requiredShippingContactFields = [.postalAddress]
+        config.requiredBillingContactFields = [.postalAddress]
+
+        let component = try? ApplePayComponent(paymentMethod: paymentMethod,
+                                               context: context,
+                                               configuration: config)
+        component?.delegate = session
+        return component
+    }
+
+    private func present(_ component: PresentableComponent) {
+        presenter?.present(viewController: component.viewController, completion: nil)
+    }
+
+    // MARK: - Alert handling
+
+    internal func presentAlert(with error: Error, retryHandler: (() -> Void)? = nil) {
+        presenter?.presentAlert(with: error, retryHandler: retryHandler)
+    }
+
+    internal func dismissAndShowAlert(_ success: Bool, _ message: String) {
+        presenter?.dismiss {
+            // Payment is processed. Add your code here.
+            let title = success ? "Success" : "Error"
+            self.presenter?.presentAlert(withTitle: title, message: message)
+        }
+    }
+
+}
+
+extension ApplePayComponentExample: AdyenSessionDelegate {
+
+    func didComplete(with resultCode: SessionPaymentResultCode, component: Component, session: AdyenSession) {
+        dismissAndShowAlert(resultCode.isSuccess, resultCode.rawValue)
+    }
+
+    func didFail(with error: Error, from component: Component, session: AdyenSession) {
+        dismissAndShowAlert(false, error.localizedDescription)
+    }
+
+    func didOpenExternalApplication(component: ActionComponent, session: AdyenSession) {}
+
+}
+
+extension ApplePayComponentExample: PresentationDelegate {
+    internal func present(component: PresentableComponent) {}
```

Comment:
Maybe good to add a comment here as well that the implementation of this delegate method is not needed

Reply:
90ede7f5ceb40b17be5ba7e0cc83f1b617a6e213